addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -0,0 +1,521 @@
+import org.openhab.io.hueemulation.internal.api.HueCreateUser;
+import org.openhab.io.hueemulation.internal.api.HueDataStore;
+import org.openhab.io.hueemulation.internal.api.HueDevice;
+import org.openhab.io.hueemulation.internal.api.HueErrorResponse;
+import org.openhab.io.hueemulation.internal.api.HueState;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Emulates A Hue compatible HTTP API server
+ *
+ * @author Dan Cunningham
+ *
+ */
+@SuppressWarnings("serial")
+public class HueEmulationServlet extends HttpServlet {
+    private Logger logger = LoggerFactory.getLogger(HueEmulationServlet.class);
+    private static final String CONFIG_PAIRING_ENABLED = "pairingEnabled";
+    private static final String PATH = "/api";
+    private static final String HOMEKIT_PREFIX = "homekit:";
+    private static final String METHOD_POST = "POST";
+    private static final String METHOD_PUT = "PUT";
+    private static final String APPLICATION_XML = "application/xml";
+    private static final String APPLICATION_JSON = "application/json";
+    private static final String CHARSET = "utf-8";
+    private static final String NEW_CLIENT_RESP = "[{\"success\":{\"username\": \"%s\"}}]";
+    private static final String STATE_RESP = "[{\"success\":{\"/lights/%s/state/on\":%s}}]";
+    private static final File USER_FILE = new File(
+            ConfigConstants.getUserDataFolder() + File.separator + "hueemulation" + File.separator + "usernames");
+    private static final File UDN_FILE = new File(
+            ConfigConstants.getUserDataFolder() + File.separator + "hueemulation" + File.separator + "udn");
+
+    /**
+     * This parses "/api/{username}/{lights}/{id}/{state}"
+     */
+    private static final Pattern PATH_PATTERN = Pattern
+            .compile(PATH + "/([^/]+)(?:(?:/(lights)/?([^/]+)?/?(state)?)?)?");
+
+    private HttpService httpService;
+    private ItemRegistry itemRegistry;
+    private EventPublisher eventPublisher;
+    private HueEmulationUpnpServer disco;
+    private String udn;
+
+    private boolean pairingEnabled = false;
+
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+        try {
+            Dictionary<String, String> servletParams = new Hashtable<String, String>();
+            httpService.registerServlet(PATH, this, servletParams, httpService.createDefaultHttpContext());
+            disco = new HueEmulationUpnpServer(PATH + "/discovery.xml", getUDN());
+            disco.start();
+            logger.info("Started Hue Emulation service at " + PATH);
+        } catch (Exception e) {
+            logger.error("Could not start Hue Emulation service: {}", e.getMessage(), e);
+        }
+    }
+
+    protected void modified(Map<String, ?> config) {
+        String pairingString = (String) config.get(CONFIG_PAIRING_ENABLED);
```

Comment:
Will do, I noticed the basic and classic ui bundles have this same pattern https://github.com/eclipse/smarthome/blob/4eb2901b52502e94beebf4293df54c60d273dbfd/extensions/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/WebAppConfig.java#L44